protocol/linkpreview/linkpreview.go

```diff
@@ -256,6 +257,21 @@ func newUnfurler(logger *zap.Logger, httpClient http.Client, url *neturl.URL) Un
 			logger:         logger,
 			httpClient:     httpClient,
 		}
+	case "giphy.com", "media.giphy.com":
+		return OEmbedUnfurler{
+			oembedEndpoint: "https://giphy.com/services/oembed",
+			url:            url,
+			logger:         logger,
+			httpClient:     httpClient,
+		}
+	case "tenor.com", "media.tenor.com":
+		// TODO: Investigate a way to unfurl previews that contain iframes (Webviews?)
```

Comment:
I would say this TODO should be removed. This question can be answered & implemented by a proper issue. The answer is yes @ibrkhalil. The old implementation used to do this ugly hack for Twitter URLs, i.e. extract metadata from the `html` field returned in the oembed response, and there's no need for iframes or webviews.

It's a sad story that certain providers like Twitter, Reddit and others don't easily expose the metadata in the oEmbed response. They really want people to inject their HTML snippets into websites so they can load their `script` tags...